README.md

```diff
@@ -166,6 +166,205 @@ This solution collects anonymous operational metrics to help AWS improve the
+
+The purpose of the endpoint is to normalize and hash clear-text PII and partition time series data sets for AMC. Timestamp columns must be formatted before using the endpoint according to ISO 8601.
+
+The `start_amc_transformation` API endpoint starts an AWS Glue Job with the arguments passed into it. The following is a summary of the arguments included in a request to the endpoint and expected outputs of the API:
+
+### Inputs
+
+- `countryCode`: string
+	- The country code to be used for different country-specific normalizations to apply to all rows in the dataset
+	- 2-digit ISO country code
+	- Ex: `"countryCode": "US"`
+	
+- `datasetId`: string
+	- The name of the new data set that will be created with the data being uploaded
+	- It can also be the name of an existing data set that the data will be uploaded to, though the schema should match (`deletedFields`, `piiFields`, and column names should be the same)
+	- Ex: `"datasetId": "new-AMC-dataset"`
+	
+- `deletedFields`: string (array)
+	- The list of columns to be deleted or dropped from the original data set prior to uploading to AMC
+	- Ex: `"deletedFields": "[\"product_quantity\",\"product_name\"]"`
+	
+	- Can be an empty array string if no data needs to be deleted
+	- Ex: `"deletedFields": "[]"`
+	
+- `destination_endpoints`: string (array)
+	- The list of AMC endpoints to receive of uploads
+	- Ex: `destination_endpoints: "[\" https://rrhcd93zj3.execute-api.us-east-1.amazonaws.com/prod\"]"`
+	
+- `outputBucket`: string
+	- The name of the ETL artifact S3 bucket where the normalized output data will be written to
+	- Ex: `"outputBucket": "myTestStack-etl-artifacts"`,
+	
+- `period`: string
+	- The time period of the data set
+	- If uploding a dimension type data set, this will be autodetected and set to `P1D`
+	- Ex: `"period": "P1D"`
+	- If a fact type data set, `autodetect` or a specific period can be selected
+		- Ex: `"period": "autodetect"`
+		- Ex: `"period": "PT1H"`
+	- Options:  One of `["autodetect","PT1M","PT1H","P1D","P7D"]`
+
+- `piiFields`: string (array)
+	- A JSON formatted list of the names of PII columns to be hashed in normalization
+	- Ex: `"piiFields": "[{\"column_name\":\"first_name\",\"pii_type\":\"FIRST_NAME\"},{\"column_name\":\"last_name\",\"pii_type\":\"LAST_NAME\"},{\"column_name\":\"email\",\"pii_type\":\"EMAIL\"}]"`
+	
+	- Can be an empty array string if no PII data is being uploaded
+	- Ex: `"piiFields": "[]"`
+	
+- `sourceBucket`: string
+	- The name of the S3 bucket where the `sourceKey` input data set is stored
+	- Ex: `"sourceKey: "myTestBucket"`
+	
+- `sourceKey`: string
+	- The S3 key of the data set to be normalized in the `sourceBucket`
+	- Ex: `"sourceKey": "myTestData.json"`
+	
+- `timestampColumn`: string
+	- Column name containing timestamps for the time series data sets
+	- Can be left as an empty string if the data set is going to be a dimension type (not time series)
```

Comment:
if dimension type it actually should not be passed in at all or an error will be thrown (if we want different behavior we should update the Glue code though)

Reply:
Hmm I grabbed these headers right off of the actual front end request to the API endpoint and it seemed fine, but ok will remove

Reply:
adding note here about our discussion:
- re evaluate what gets passed from UI to API in dimension vs fact datasets
- provide argument checking in the API rather than Glue